src/realm/object-store/sync/sync_session.hpp

```diff
@@ -427,7 +436,10 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     mutable util::CheckedMutex m_config_mutex;
     RealmConfig m_config GUARDED_BY(m_config_mutex);
     const std::shared_ptr<DB> m_db;
-    const std::shared_ptr<sync::SubscriptionStore> m_flx_subscription_store;
+    std::shared_ptr<sync::SubscriptionStore> m_flx_subscription_store GUARDED_BY(m_config_mutex);
```

Comment:
I think there are places where the store is accessed without the mutex being locked

Reply:
The only places where it's used directly are in `create_sync_session()`, which locks the config mutex and `update_subscription_store()` which requires the config mutex to be locked before calling. The other instances use the `get_flx_subscription_store()` which grabs the config mutex and returns a shared_ptr.

Reply:
Unfortunately, `update_subscription_store()` is being called in the constructor, and I was getting TSAN errors (see [here](https://parsley.mongodb.com/evergreen/realm_core_stable_ubuntu2004_tsan_object_store_tests_patch_5b2fc04262d873909f0a2c62802cf0805c50d6a3_6411c57a7742ae4ff6061600_23_03_15_13_17_47/0/task?shareLine=5622)) when locking the mutexes in the constructor, so I don't think I will be able to update the function to take the mutex objects...